packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -407,9 +574,50 @@ async function selectPlatformAndExecute(
           previewExecution.command.toString()
         );
         vscode.window.showInformationMessage(
-          nls.localize('force_lightning_lwc_android_start', target)
+          nls.localize('force_lightning_lwc_android_start', targetDevice)
         );
       }
     });
   }
 }
+
+/**
+ * Given a path, it recursively goes through that directory and upwards, until it finds
+ * a config file named sfdx-project.json and returns the path to the folder containg it.
+ *
+ * @param startPath starting path to search for the config file.
+ * @returns the path to the folder containing the config file, or undefined if config file not found
+ */
+function getProjectRootDirectory(startPath: string): string | undefined {
```

Comment:
We are currently only supporting SFDX project in a single workspace folder. So for most of our existing implementation look up the first folder in the workspace, but this is fine to me too. https://github.com/forcedotcom/salesforcedx-vscode/blob/1a3987f593eb053c5d461ef38beb73c28ae37c04/packages/salesforcedx-vscode-core/src/util/rootWorkspace.ts#L21

Reply:
Yes, we discussed in our team and thought that if we go about doing it the way I'm doing here, then in the future when we support multiple workspaces then this code should work too. So if this is fine by you then we'll leave it as is.